activation/activation.go

```diff
@@ -477,14 +477,17 @@ func (b *Builder) UpdatePoETServers(ctx context.Context, endpoints []string) err
 
 	clients := make([]PoetProvingServiceClient, 0, len(endpoints))
 	for _, endpoint := range endpoints {
-		client := b.poetClientInitializer(endpoint, b.poetCfg)
+		client, err := b.poetClientInitializer(endpoint, b.poetCfg)
+		if err != nil {
+			return &PoetSvcUnstableError{source: fmt.Errorf("failed to initialize poet client for '%s': %w", endpoint, err)}
```

Comment:
nit: i would shorten this. maybe just "init poet client %s: %w", here and other places.